src/gwt/src/org/rstudio/studio/client/workbench/prefs/views/SpellingPreferencesPane.java

```diff
@@ -54,12 +57,28 @@ public SpellingPreferencesPane(GlobalDisplay globalDisplay,
       spaced(customDictsWidget_);
       nudgeRight(customDictsWidget_);
       add(customDictsWidget_);
-            
+
       add(checkboxPref("Ignore words in UPPERCASE", prefs.ignoreUppercaseWords()));
-      
       add(checkboxPref("Ignore words with numbers", prefs.ignoreWordsWithNumbers()));
 
-      add(checkboxPref("Use real time spellchecking", prefs.realTimeSpellchecking()));
+      boolean canRealtime = TypoSpellChecker.canRealtimeSpellcheckDict(prefs.spellingDictionaryLanguage().getValue());
+      realtimeSpellcheckingCheckbox_ = checkboxPref("Use real time spellchecking", prefs.realTimeSpellchecking());
+      realtimeSpellcheckingCheckbox_.getElement().getStyle().setColor(canRealtime ? "black" : "darkgrey");
+      add(realtimeSpellcheckingCheckbox_);
+
+      blacklistWarning_ = new Label("Real time spellchecking currently unavailable for this dictionary");
+      blacklistWarning_.getElement().getStyle().setColor("red");
+      blacklistWarning_.setVisible(!canRealtime);
+
+      add(blacklistWarning_);
+
+      languageWidget_.addChangeHandler((event) -> {
+         boolean canRealtimeCheck = TypoSpellChecker.canRealtimeSpellcheckDict(languageWidget_.getSelectedLanguage());
+         blacklistWarning_.setVisible(!canRealtimeCheck);
+         realtimeSpellcheckingCheckbox_.setValue(realtimeSpellcheckingCheckbox_.getValue() && canRealtimeCheck);
+         realtimeSpellcheckingCheckbox_.setEnabled(canRealtimeCheck);
+         realtimeSpellcheckingCheckbox_.getElement().getStyle().setColor(canRealtimeCheck ? "black" : "darkgrey");
```

Comment:
There are a number of RStudio themes in the wild which attempt to dark-theme the dialogs. This is not technically supported but this will break them. What do you think about using opacity instead?

Reply:
I'll give it a shot and see if it looks ok. Would be curious to play around with those themes!

Reply:
It looks good!
![ncpqAbB](https://user-images.githubusercontent.com/136863/73384465-51eae700-4299-11ea-9d6b-1c2b36cabd9b.png)

Reply:
Here's one such theme: https://github.com/anthonynorth/rscodeio